runtime/jsMain/src/kotlinx/serialization/DynamicObjectParser.kt

```diff
@@ -14,7 +15,10 @@ import kotlin.math.floor
  */
 internal const val MAX_SAFE_INTEGER: Double = 9007199254740991.toDouble() // 2^53 - 1
 
-class DynamicObjectParser(context: SerialModule = EmptyModule): AbstractSerialFormat(context) {
+class DynamicObjectParser(
+    context: SerialModule = EmptyModule,
+    internal val configuration: JsonConfiguration = JsonConfiguration.Default
```

Comment:
This may be a little weird, but JSON is the closest approximation to JS objects so if you squint it doesn't hurt. At least not as bad as duplicating all of the controls? Although I can pursue that as well if you'd like.

Reply:
Well, it makes sense in a light of #628 , where 'encodeDefaults' and 'allowStructuredMapKeys' could also be used (although structured map keys have to be supported in Parser, too). It's also possible to implement strict mode in parser, however, no one complained about lack of it yet.